dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs

```diff
@@ -146,6 +146,41 @@ public async IAsyncEnumerable<DatabaseEntry> ReadAllAsync(SqliteConnection conn,
         return null;
     }
 
+    public async IAsyncEnumerable<DatabaseEntry> ReadBatchAsync(SqliteConnection conn,
+    string collectionName,
+    string[] keys,
+    bool withEmbeddings = false,
+    [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        using SqliteCommand cmd = conn.CreateCommand();
+        var keyParameters = keys.Select((key, index) => $"@key{index}");
+        var parameters = string.Join(", ", keyParameters);
+
+        var selectFieldQuery = withEmbeddings ? "*" : "key, metadata, timestamp";
+#pragma warning disable CA2100 // Review SQL queries for security vulnerabilities
```

Comment:
Suggest using parameterization to avoid SQL-Injection vulnerabilities (remove the pragma as well)

Reply:
@RogerBarreto I followed the SqlServerClient implementation. It doesn't support select column as parameterization. 
What will be the best approach in this kind of scenario?